staff/views.py

```diff
@@ -55,40 +62,27 @@ def classroom_management(request):
     classroom_list = Classroom.objects.all()
     for classroom in classroom_list:
         class_dict.update({classroom.course: get_class_member_dict(classroom)})
-    return render(request, "classroom_management.html", {"classrooms": classroom_list,
-                                                         "class_dicts": class_dict})
+    return render(
+        request,
+        "classroom_management.html",
+        {"classrooms": classroom_list, "class_dicts": class_dict},
+    )
 
 
 @group_required("staff")
 def create_staff_user(request):
     if request.method == "POST":
         form = CreateStaffForm(request.POST)
         if form.is_valid():
-            form.save()
-            random_password = DjangoUser.objects.make_random_password()
-            new_user = create_user_with_profile(form, random_password)
-            email = form.cleaned_data.get("email")
-            staff_group = Group.objects.get(name="staff")
-            staff_group.user_set.add(new_user)
-            UserSocialAuth.objects.create(
-                uid=form.cleaned_data.get("email"),
-                user_id=new_user.userprofile.user_id,
-                provider="google-oauth2",
-            )
-            first_name = form.cleaned_data.get("first_name")
-            messages.success(
-                request, f"Staff Account Successfully Created For {first_name}"
-            )
-            email_new_user(
-                request, email, first_name, "staff", new_user.username, random_password
-            )
+            save_user_to_salesforce(request, form)
+            create_mission_bit_user(request, form, "staff")
             return redirect("staff")
         else:
             messages.error(
                 request,
                 "Staff User NOT created, your form was not valid, please try again.",
             )
-            return redirect("staff")
+            return redirect("create_staff_user")
```

Comment:
Yes I was using redirect improperly.  I looked it up and now I am using render_to_response() and reloading the page with appropriate error messages.  I will have this done for the whole app by tomorrow.  

Regarding workflow here, if this task weren't already inside a pull request, would you normally want a pull request to be of this scale?  I've definitely had a fundamental misconception of what pull requests should be driven by (i.e. I have basically putting one week's worth of work into each one, to avoid creating too many, while I should be thinking of each pull request as associated with a certain task).  While we're in this current pull request though, a lot more work is going to be generated for everyone, so I'm not sure if there's a way to implement a new paradigm before this pull request is complete.  Let me know what you think... and again, sorry for the enormity of this one.

Reply:
It’s up to you whether to do that separately or not. The damage is already done really. Another option is to merge more or less what we have now and fix everything in smaller PRs. We don’t have production deployments to worry about so master’s state isn’t a big issue. 

Another factor that made this PR bigger than it should’ve been was reformatting all of the code. That should’ve been done in a PR of its own, and ideally done before each commit from then on to avoid reformatting code that hasn’t changed.

Reply:
I'll reformat once more before we merge and then before every commit from there on.